spec/rubocop/cop/lint/duplicate_set_element_spec.rb

```diff
@@ -0,0 +1,157 @@
+      Set[:foo, :bar]
+    RUBY
+  end
+
+  it 'registers an offense when using duplicate symbol element in `::Set[]`' do
+    expect_offense(<<~RUBY)
+      ::Set[:foo, :bar, :foo]
+                        ^^^^ Remove the duplicate element in Set.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      ::Set[:foo, :bar]
+    RUBY
+  end
+
+  it 'registers an offense when using multiple duplicate symbol element' do
+    expect_offense(<<~RUBY)
+      Set[:foo, :bar, :foo, :baz, :baz]
+                      ^^^^ Remove the duplicate element in Set.
+                                  ^^^^ Remove the duplicate element in Set.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      Set[:foo, :bar, :baz]
+    RUBY
+  end
+
+  it 'registers an offense when using duplicate lvar element' do
+    expect_offense(<<~RUBY)
+      foo = do_foo
+      bar = do_bar
+
+      Set[foo, bar, foo]
+                    ^^^ Remove the duplicate element in Set.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      foo = do_foo
+      bar = do_bar
+
+      Set[foo, bar]
+    RUBY
+  end
+
+  it 'registers an offense when using duplicate ivar element' do
+    expect_offense(<<~RUBY)
+      Set[@foo, @bar, @foo]
+                      ^^^^ Remove the duplicate element in Set.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      Set[@foo, @bar]
+    RUBY
+  end
+
+  it 'registers an offense when using duplicate constant element' do
+    expect_offense(<<~RUBY)
+      Set[Foo, Bar, Foo]
+                    ^^^ Remove the duplicate element in Set.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      Set[Foo, Bar]
+    RUBY
+  end
+
+  it 'does not register an offense when using duplicate method call element' do
+    expect_no_offenses(<<~RUBY)
+      Set[foo, bar, foo]
+    RUBY
+  end
+
+  it 'does not register an offense when using duplicate safe navigation method call element' do
+    expect_no_offenses(<<~RUBY)
+      Set[obj&.foo, obj&.bar, obj&.foo]
+    RUBY
+  end
+
+  it 'registers an offense when using duplicate symbol element in `Set.new([...])`' do
+    expect_offense(<<~RUBY)
+      Set.new([:foo, :bar, :foo])
+                           ^^^^ Remove the duplicate element in Set.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      Set.new([:foo, :bar])
+    RUBY
+  end
+
+  it 'registers an offense when using duplicate symbol element in `Set.new(%i[...])`' do
+    expect_offense(<<~RUBY)
+      Set.new(%i[foo bar foo])
+                         ^^^ Remove the duplicate element in Set.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      Set.new(%i[foo bar])
+    RUBY
+  end
+
+  it 'registers an offense when using duplicate symbol element in `::Set.new([...])`' do
+    expect_offense(<<~RUBY)
+      ::Set.new([:foo, :bar, :foo])
+                             ^^^^ Remove the duplicate element in Set.
+    RUBY
+
+    expect_correction(<<~RUBY)
+      ::Set.new([:foo, :bar])
+    RUBY
+  end
+
+  it 'does not register an offense when not using duplicate element in Set' do
```

Comment:
This tests method calls, so will not register an offense because of that if I understand correctly.

Reply:
The test is written that way, but is there any problem with it?

Reply:
If it is intended that way then the test description should mention method calls, since it would not register an offense even if there was duplication with method calls, I think.